core/src/main/java/fr/sncf/osrd/envelope/EnvelopePhysics.java

```diff
@@ -174,4 +174,44 @@ public static double intersectStepWithSpeed(
             interpolatedPosition = minPos;
         return interpolatedPosition;
     }
+
+    /** Returns if two EnvelopeParts intersect */
```

Comment:
this function doesn't really have a fair description: it does not detect all intersections, as it iterates on a and interpolates on b. if a has like two far apart points and b is high frequency, it will only detect an intersection if the sign of startSpeedA - b.interpolateSpeed(startPosA) isn't the same as endSpeedA - b.interpolateSpeed(endPosA)